src/audio/module_adapter/module/volume/volume.c

```diff
@@ -436,6 +439,7 @@ static int volume_init(struct processing_module *mod)
 
 	cd->ramp_type = vol->ramp;
 	cd->initial_ramp = vol->initial_ramp;
+	cd->copy_needed = true;
```

Comment:
Since you set it at function reset_state(), and the function will be called at the end of volume_init, do not need to set it again here.

Reply:
reset adding is needed, however, reset_state also called from volume_reset, let me delete the setting in volume_init.